src/Config/HotReloadEventHandler.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config
+{
+    public class HotReloadEventHandler<TEventArgs> where TEventArgs : CustomEventArgs
+    {
+        public event EventHandler<TEventArgs>? QueryManagerFactory_ConfigChangeEventOccurred;
```

Comment:
Good suggestion, need to ensure the members of this dictionary are populated accurately in respective constructors

Reply:
I added this, very nice form, though I couldn't get nameof to work inside of this class because there are no references to those names so I just used hard coded strings.